docs/understanding-your-pipeline/canonical-event/index.md

```diff
@@ -36,274 +42,490 @@ The Snowplow data table is designed to be immutable: the data in each line shoul
+</TabItem>
+<TabItem value="bq" label="BigQuery">
 
-#### Custom unstructured events
+The unstructured event is not part of the standard `atomic.events` columns, it is instead a `RECORD` type column with the name of the schema and the full version number. The fields in this record will be determined by the JSON schema defined for the event in advance, converted to snake case. Users can query fields in the self-describing event by extracting them using 
 
-Custom unstructured events are a flexible tool that enable Snowplow users to define their own event types and send them into Snowplow.
+```sql
+select
+...
+unstruct_event_my_example_event_1_0_0.my_custom_field,
+...
+from 
+    atomic.events
+```
+
+</TabItem>
+<TabItem value="sf" label="Snowflake">
+
+The unstructured event is not part of the standard `atomic.events` columns, it is instead a `OBJECT` type column with the name of the schema and the major version number. The fields in this object will be set by the JSON schema defined for the event in advance, but are not a fixed structure like in other warehouses. Users can query fields in the self-describing event by extracting them using 
+
+```sql
+select
+...
+unstruct_event_my_example_event_1_0_0:myCustomField::varchar, -- field will be variant type so important to cast
+...
+from 
+    atomic.events
+```
+
+</TabItem>
+<TabItem value="db" label="Databricks">
+
+The unstructured event is not part of the standard `atomic.events` columns, it is instead a `struct` type column with the name of the schema and the major version number. The fields in this table will be determined by the JSON schema defined for the event in advance, converted to snake case. Users can query fields in the self-describing event by extracting them using 
 
-When a user sends in a custom unstructured event, they do so as a JSON of name-value properties that conforms to a JSON schema defined for the event earlier.
+```sql
+select
+...
+unstruct_event_my_example_event_1_0_0.myCustomField,
+...
+from 
+    atomic.events
+```
+
+</TabItem>
+</Tabs>
+
+
+#### Entities (Contexts)
 
-The unstructured event is not part of the `atomic.events` table; instead, for users running on Redshift, it is shredded into its own table. The fields in this table will be determined by the JSON schema defined for the event in advance. Users can query just the table for that particular unstructured event, if that's all that's required for their analysis, or join that table back to the `atomic.events` table by
+Entities (Contexts) enable Snowplow users to define their own entities that are related to events, and fields that are related to each of those entities. For example, an online retailer may choose to define a `user` context, to store information about a particular user, which might include data points like the users Facebook ID, age, membership details etc. In addition, they may also define a `product` context, with product data e.g. SKU, name, created date, description, tags etc.
```

Comment:
Maybe bring this home with an event to complete the picture? e.g. "product_view" shows that the specific user viewed the specific product.